pkg/hubble/exporter/config_watcher.go

```diff
@@ -57,69 +66,33 @@ func (c *configWatcher) watch(ctx context.Context, interval time.Duration) error
 }
 
 func (c *configWatcher) reload() {
-	c.logger.Debug("Attempting reload")
-	config, hash, err := c.readConfig()
+	// TODO: should we keep track of the previous hash and avoid parsing/invoking the callback?
+	// See if the metric DynamicExporterConfigLastApplied needs to be updated on all reloads
+	// or only when config changes. If we want to track reloads in general, consider introducing
+	// a label for whether an update was needed or not.
+	configs, hash, err := c.parseConfig()
```

Comment:
I'd appreciate any inputs on this.

Reply:
IMO at this level `reload()` should just do reload, and the callback may decide what to do with the hash / config information. No strong opinion on this though, maybe @chancez has one.

Reply:
Sounds good, for now I removed the TODO